jax/numpy/linalg.py

```diff
@@ -367,3 +367,46 @@ def solve(a, b):
 for func in get_module_functions(onp.linalg):
   if func.__name__ not in globals():
     globals()[func.__name__] = _not_implemented(func)
+
+
+@_wraps(onp.linalg.lstsq, lax_description=textwrap.dedent("""\
+    It has two important differences:
+
+    1. In `numpy.linalg.lstsq`, the default `rcond` is `-1`, and shows a deprecation
+       warning. In jax.numpy, the default rcond is `None`.
+    2. In `np.linalg.lstsq` the residuals return an empty list for low-rank solutions.
+       Here, the residuals are returned in all cases, to make the function compatible
+       with jit. The non-jit compatible numpy behavior can be recovered by passing
+       numpy_resid=True.
+    """))
+def lstsq(a, b, rcond=None, *, numpy_resid=False):
+  # TODO: add lstsq to lax_linalg and implement this function via those wrappers.
+  a, b = _promote_arg_dtypes(a, b)
+  dtype = a.dtype
+  b_ndim = b.ndim
+  if b_ndim == 1:
+    b = b[:, None]
```

Comment:
I go back and forth on this, but arguably `newaxis` is more readable:
```suggestion
    b = b[:, np.newaxis]
```

Reply:
Done